src/us/kbase/workspace/test/database/provenance/FundingReferenceTest.java

```diff
@@ -21,163 +23,229 @@
+					.withFunderID(entry.getKey())
+					.withAwardID(AWARD_ID_STRING_UNTRIMMED)
+					.withAwardTitle(AWARD_TITLE_STRING_UNTRIMMED)
+					.withAwardURL(AWARD_URL_STRING_UNTRIMMED)
+					.build();
+			final Map<String, String> expectedMap = ImmutableMap.of(
+				FUNDER_NAME, FUNDER_NAME_STRING,
+				FUNDER_ID, entry.getValue(),
+				AWARD_ID, AWARD_ID_STRING,
+				AWARD_TITLE, AWARD_TITLE_STRING);
+			assertFundingReferenceFields(fr, expectedMap, new URL(AWARD_URL_STRING));
 		}
 	}
 
-
 	@Test
 	public void buildWithNullOrWhitespace() throws Exception {
 		for (String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
-			final FundingReference fr = FundingReference.getBuilder(FUNDER_NAME)
-				.withFunderID(nullOrWs)
-				.withAwardID(nullOrWs)
-				.withAwardTitle(nullOrWs)
-				.withAwardURL(nullOrWs)
-				.build();
-			assertThat(INCORRECT_FUNDER_NAME, fr.getFunderName(), is(FUNDER_NAME));
-			assertThat(INCORRECT_FUNDER_ID, fr.getFunderID(), is(ES));
-			assertThat(INCORRECT_AWARD_ID, fr.getAwardID(), is(ES));
-			assertThat(INCORRECT_AWARD_TITLE, fr.getAwardTitle(), is(ES));
-			assertThat(INCORRECT_AWARD_URL, fr.getAwardURL(), is(ES));
+			final FundingReference fr = FundingReference.getBuilder(FUNDER_NAME_STRING)
+					.withFunderID(nullOrWs)
+					.withAwardID(nullOrWs)
+					.withAwardTitle(nullOrWs)
+					.withAwardURL(nullOrWs)
+					.build();
+			assertFundingReferenceFields(fr, MINIMAL_MAP, null);
 		}
 	}
 
 	@Test
 	public void buildAndOverwrite() throws Exception {
 		for (String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
-			final FundingReference fr = FundingReference.getBuilder(FUNDER_NAME)
-				.withFunderID(FUNDER_ID).withFunderID(nullOrWs)
-				.withAwardID(AWARD_ID).withAwardID(nullOrWs)
-				.withAwardTitle(AWARD_TITLE).withAwardTitle(nullOrWs)
-				.withAwardURL(AWARD_URL).withAwardURL(nullOrWs)
-				.build();
-			assertThat(INCORRECT_FUNDER_NAME, fr.getFunderName(), is(FUNDER_NAME));
-			assertThat(INCORRECT_FUNDER_ID, fr.getFunderID(), is(ES));
-			assertThat(INCORRECT_AWARD_ID, fr.getAwardID(), is(ES));
-			assertThat(INCORRECT_AWARD_TITLE, fr.getAwardTitle(), is(ES));
-			assertThat(INCORRECT_AWARD_URL, fr.getAwardURL(), is(ES));
+			final FundingReference fr = FundingReference.getBuilder(FUNDER_NAME_STRING)
+					.withFunderID(FUNDER_ID_STRING).withFunderID(nullOrWs)
+					.withAwardID(AWARD_ID_STRING).withAwardID(nullOrWs)
+					.withAwardTitle(AWARD_TITLE_STRING).withAwardTitle(nullOrWs)
+					.withAwardURL(AWARD_URL_STRING).withAwardURL(nullOrWs)
+					.build();
+			assertFundingReferenceFields(fr, MINIMAL_MAP, null);
+		}
+	}
+
+	/**
+	 * Checks the error thrown when a FundingReference is built with invalid
+	 * arguments
+	 *
+	 * @param builder
+	 *                funding reference builder, complete with args
+	 * @param error
+	 *                the expected error
+	 */
+	private void buildFundingRefFailWithError(final FundingReference.Builder builder, final String errorString) {
```

Comment:
refactor again - function for checking error message for failing object construction